homeassistant/components/mqtt/mixins.py

```diff
@@ -1130,21 +1130,33 @@ def _set_entity_name(self, config: ConfigType) -> None:
             # Assign the default name
             self._attr_name = self._default_name
         if CONF_DEVICE in config:
+            device_name: str
             if CONF_NAME not in config[CONF_DEVICE]:
                 _LOGGER.info(
                     "MQTT device information always needs to include a name, got %s, "
                     "if device information is shared between multiple entities, the device "
                     "name must be included in each entity's device configuration",
                     config,
                 )
-            elif config[CONF_DEVICE][CONF_NAME] == entity_name:
+            elif (device_name := config[CONF_DEVICE][CONF_NAME]) == entity_name:
                 _LOGGER.warning(
                     "MQTT device name is equal to entity name in your config %s, "
                     "this is not expected. Please correct your configuration. "
                     "The entity name will be set to `null`",
                     config,
                 )
                 self._attr_name = None
+            elif isinstance(entity_name, str) and entity_name.startswith(device_name):
+                new_entity_name = entity_name[len(device_name) :].lstrip()
+                _LOGGER.warning(
+                    "MQTT entity name starts with the device name in your config %s, "
+                    "this is not expected. Please correct your configuration. "
+                    "The device name prefix will be stripped off the entity name "
+                    "and becomes '%s'",
+                    config,
+                    new_entity_name,
+                )
+                self._attr_name = new_entity_name
```

Comment:
The resulting entity name is lower case, but should start with an uppercase letter.
`self._attr_name = new_entity_name.title()`

Reply:
That would turn all other chars to lower, my PR only converts the first char if the device name is a capital.